plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XProjectManager.java

```diff
@@ -197,6 +197,10 @@ protected XBuildResult doBuild(Set<URI> dirtyFiles, Set<URI> deletedFiles,
 
 		XBuildResult result = incrementalBuilder.build(request);
 
+		if (!request.isGeneratorEnabled()) {
```

Comment:
I don't really like the flow from WorkspaceManager->BuildManager->ProjectManager->WorkspaceManager

Can we maintain the list of waiting files at the same location where we maintain the pending deltas?

Reply:
Good idea! Done.